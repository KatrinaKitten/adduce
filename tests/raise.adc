; Raising and catching errors

Do (
  Catch (
    Let Message be the error message.
    Print the Concatenate of 'Error: ' with Message.
  ).

  Raise "Same level".
  Do (Raise "Lower level").

  Do (
    Catch (Print "Caught by the inner catch". Drop).
    Raise "Not the same message".
  ).

  Print "This should execute".
  Do (
    Raise "Some other message".
    Print "This should not execute".
  ).

  Do (
    Catch (Print "Inner catch with re-raise". Raise).
    Raise "It works!".
  ).
).

