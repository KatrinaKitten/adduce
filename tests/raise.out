Error: Same level
Error: Lower level
Caught by the inner catch
This should execute
Error: Some other message
Inner catch with re-raise
Error: It works!
